utils_cv/detection/model.py

```diff
@@ -23,54 +35,59 @@
 from ..common.gpu import torch_device
 
 
-def _get_det_bboxes(
-    pred: List[dict], labels: List[str], im_path: str = None
-) -> List[DetectionBbox]:
-    """ Gets the bounding boxes and labels from the prediction object
+def _get_det_bboxes_and_mask(
+    pred: Dict, labels: List[str], im_path: Union[str, Path] = None
+) -> Dict:
+    """ Gets the bounding boxes and masks from the prediction object.
 
     Args:
         pred: the output of passing in an image to torchvision's FasterRCNN
-        model
-        labels: list of labels
+            or MaskRCNN model
+        labels: list of labels without "__background__".
         im_path: the image path of the preds
-
     Return:
-        a list of DetectionBboxes
+        a dict of DetectionBboxes and masks
     """
-    pred_labels = pred[0]["labels"].detach().cpu().numpy().tolist()
-    pred_boxes = (
-        pred[0]["boxes"].detach().cpu().numpy().astype(np.int32).tolist()
-    )
-    pred_scores = pred[0]["scores"].detach().cpu().numpy().tolist()
-
-    det_bboxes = []
-    for label, box, score in zip(pred_labels, pred_boxes, pred_scores):
-        label_name = labels[label - 1]
-        det_bbox = DetectionBbox.from_array(
-            box,
-            score=score,
-            label_idx=label,
-            label_name=label_name,
-            im_path=im_path,
-        )
-        det_bboxes.append(det_bbox)
 
-    return det_bboxes
+    label_names = np.array(labels)[pred['labels'] - 1]
```

Comment:
The function looks nice since statements like "pred_labels = pred[0]["labels"].detach().cpu().numpy().tolist()" got removed. But now it's assumed that the _apply_threshold() function HAS to be called, since in there the detach() etc is called. This is not nice. My suggestion: change back to the original code and remove the from_arrays() function since I guess it's not used anymore.

Reply:
Reverted and removed `from_arrays()`